near-plugins/tests/common/utils.rs

```diff
@@ -146,3 +160,44 @@ pub fn assert_failure_with(res: ExecutionFinalResult, must_contain: &str) {
         err,
     );
 }
+
+/// Returns the block timestamp in nanoseconds. Panics on failure.
+async fn block_timestamp(worker: &Worker<Sandbox>) -> u64 {
+    worker
+        .view_block()
+        .await
+        .expect("Should view block")
+        .timestamp()
+}
+
+/// Returns the block in which a transaction or receipt was included.
+pub async fn get_transaction_block(
+    worker: &Worker<Sandbox>,
+    result: &ExecutionOutcome,
+) -> workspaces::Result<Block> {
+    let block_hash = result.block_hash;
+    worker.view_block().block_hash(block_hash).await
+}
+
+/// [Time travels] `worker` forward by at least `duration`. This is achieved by a very naive
+/// approach: fast forward blocks until `duration` has passed. Keeping it simple since this function
+/// is available only in tests.
+///
+/// Due to this approach, it is recommended to pass only relatively small values as `duration`. Fast
+/// forwarding provided by this function is reasonly fast in our tests for durations that correspond
+/// to less than 100 seconds.
+///
+/// [Time travels]: https://github.com/near/workspaces-rs#time-traveling
+pub async fn fast_forward_beyond(worker: &Worker<Sandbox>, duration: Duration) {
+    let initial_timestamp = block_timestamp(worker).await;
+
+    // Estimating a number of blocks to skip based on `duration` and calling `fast_forward` only
+    // once seems more efficient. However, that leads to jittery tests as `fast_forward` may _not_
+    // forward the block timestamp significantly.
```

Comment:
I feel like there's a middle ground here where we can fast-forward in larger steps until the duration to wait is smaller. For example, if we are skipping 10s into the future then it seems safe that we could skip 5 blocks and then check how much farther we need to go.

But then again, this is only used in tests so you're probably right that it is not worth the additional complexity for a minor speed-up. Let's leave it as-is unless it becomes a problem.

Reply:
It’s tricky because we need to move forward in time but `fast_forward` moves forward in blocks. Since forwarding by multiple blocks still may _not_ forward the timestamp, I guess finding that middle ground is difficult. That’s why I went for this very naive approach.